src/ShopifyApp/Services/ShopSession.php

```diff
@@ -287,38 +287,38 @@ public function setCookiePolicy()
      */
     private function checkSameSiteNoneCompatible()
     {
-        $compatible = false;
+        $compatible = true;
```

Comment:
Why would you switch this?

If you start with the default of compatible then random browsers that do not pass the 'fail' checks will be marked as compatible and will break.

For example, browsing from lynx in the terminal, using the playstation or switch browser, random internet connected fridges and cars etc.

It's better to start off as incompatible, and then only mark the ones that are definitely compatible.

Reply:
So far the examples from both Chrome (https://www.chromium.org/updates/same-site/incompatible-clients) and Shopify (https://github.com/shopify/shopify_app/pull/851/) default as compatible and only mark as incompatible in the scenario that it is known to not work. I feel that it's wiser to adhere to their strategy than to be counter to it.

I believe the reasoning is because older browsers will ignore this flag and treat it as the default setting for a cookie, and managing a list of "compatible" browsers is only going to be harder to maintain with time as more and more browsers (such as Firefox and MS Edge) begin to enforce stricter cookie protections. In fact, the list of incompatible browsers Google has identified are browsers that behave unexpectedly when the flag is passed. That is why iOS 12 is incompatible, but iOS 11 is compatible (it simply ignores the flag).

Reply:
Actually, now that I think about it, technically the only browser that supports `SameSite=None` is Chrome. Neither UCBrowser nor Safari support that flag even now. So by that theory, only Chrome should be marked as compatible. 

So for Google to recommend only blacklisting browsers that are known to break a site with that flag, it means that all other browsers should be assumed will ignore that flag and considered "compatible".

Reply:
I wouldn't take that example from Chrome as gospel. They will of course assume their way is correct and forget about things like crawlers, scripts that use cookie jars etc. SameSite=none isn't even a standard yet, it's still in draft.

Perhaps they're trying to use their large market share to force the change to become standard. If everyone starts sending SameSite=none; Secure back to all browsers it's easier for them to get the standard to pass.

I personally believe it's best to assume that everything is _not_ compatible with SameSite=none, and add browsers that are compatible with this new Chrome standard over time (if any).

However, I do get what you're saying about obscure browsers possibly just ignoring the SameSite=none flag and treating it the same as SameSite=lax

That being said, I'm happy with either approach, as long as it ensures all of our apps continue to work.

Reply:
I get where you're coming from @darrynten. 

For the sake of our apps continuing to work without having to play catch-up as Edge and Firefox adopts stricter `SameSite=None` rules (both vendors have already agreed to follow suit, but have no schedule yet), I think this is the better option. Besides, since Shopify's official Rails library has adopted the black-list strategy (https://github.com/Shopify/shopify_app/pull/851), if we get this wrong, so would have all of our fellow Rails peers. So I think we are safe ;)